packages/common/container-definitions/src/loader.ts

```diff
@@ -102,23 +102,134 @@ export interface ICodeAllowList {
 }
 
 /**
- * Events emitted by the Container "upwards" to the Loader and Host
+ * Events emitted by the {@link IContainer} "upwards" to the Loader and Host.
  */
+/* eslint-disable @typescript-eslint/unified-signatures */
 export interface IContainerEvents extends IEvent {
+    /**
+     * Emitted when the readonly state of the container changes.
+     *
+     * @remarks Listener parameters:
+     *
+     * - `readonly`: Whether or not the container is now in a readonly state.
+     *
+     * @see {@link IContainer.readOnlyInfo}
+     */
     (event: "readonly", listener: (readonly: boolean) => void): void;
+
+    /**
+     * Emitted when the {@link IContainer} completes connecting to the Fluid service.
+     *
+     * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+     *
+     * @see
+     *
+     * - {@link IContainer.connectionState}
+     *
+     * - {@link IContainer.connect}
+     */
     (event: "connected", listener: (clientId: string) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `codeDetails`: TODO
+     *
+     * - `proposal`: TODO
+     *
+     * @see {@link IContainer.proposeCodeDetails}
+     */
     (event: "codeDetailsProposed", listener: (codeDetails: IFluidCodeDetails, proposal: ISequencedProposal) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `codeDetails`: TODO
+     */
     (event: "contextChanged", listener: (codeDetails: IFluidCodeDetails) => void);
-    (event: "disconnected" | "attached", listener: () => void);
+
+    /**
+     * Emitted when the {@link IContainer} becomes disconnected from the Fluid service.
+     *
+     * @remarks Reflects connection state changes against the (delta) service acknowledging ops/edits.
+     *
+     * @see
+     *
+     * - {@link IContainer.connectionState}
+     *
+     * - {@link IContainer.disconnect}
+     */
+    (event: "disconnected", listener: () => void);
+
+    /**
+     * TODO
+     *
+     * @see {@link IContainer.attachState}
+     */
+    (event: "attached", listener: () => void);
+
+    /**
+     * Emitted when the {@link IContainer} is closed, which permanently disables it.
+     *
+     * @remarks Listener parameters:
+     *
+     * - `error`: If the container was closed due to error, this will contain details about the error that caused it.
+     */
     (event: "closed", listener: (error?: ICriticalContainerError) => void);
+
+    /**
+     * TODO
```

Comment:
Original intention (I believe) of this event was to emit warnings (indication of some sort of state of container that may lead to errors) that are actionable to the consumer of IContainer. In our codebase that is only throttling at this point, but I know FFX is using this path to push some of their own warnings up. We were looking to deprecate this path at some point.

Reply:
What is the intended replacement? Should we mark this as deprecated now?

Reply:
@ssimic2 also is there a good point of contact for further clarity on this?

Reply:
warnings should be removed - no replacement. We should find a way to show proper information on each API. I.e. if it's throttling on blob download, then blob download API should expose this info somehow. There are some tasks tracking this work, but not plan to make them in reality yet.

Reply:
@vladsud Just to be clear, are you suggesting we mark as deprecated immediately? Or do we need to pursue those other tasks first? If I do mark as deprecated, I would like to make note of any related plans.

Reply:
I'd check Bohemia code base - if they do not use it, then yes, mark as deprecated. If they use it, we should not touch it and have a better long term plan implemented before marking / removing it.

Reply:
There appear to be 2 usages currently. I added a note indicating we intend to replace this in the future, but I've left it un-deprecated for the time being.

Reply:
Feel free to re-open this if you would like to see any changes / clarifications made in the note.